docs/build_meta.txt

```diff
@@ -0,0 +1,95 @@
+=======================================
+Documentation to setuptools.build_meta
+=======================================
+
+What is it?
+-------------
+
+Python packaging has come `a long way <https://www.bernat.tech/pep-517-518/>`_.
+
+The traditional ``setuptools``'s way of packgaging Python modules
+uses a ``setup()`` function within the ``setup.py`` script. Commands such as
+``python setup.py bdist`` or ``python setup.py bdist_wheel`` generate a 
+distribution bundle and ``python setup.py install`` installs the distribution. 
+This interface makes it difficult to choose other packaging tools without an 
+overhaul. Additionally, the ``setup.py`` scripts hasn't been the most user
+friendly tool.
+
+PEP517 therefore came to rescue and specified a new standard to 
+package and distribute Python modules. Under PEP517:
+
+    a ``pyproject.toml`` file is used to specify what program to use
+    for generating distribution. 
+
+    Then, two functions provided by the program, ``build_wheel(directory: str)`` 
+    and ``build_sdist(directory: str)`` create the distribution bundle at the 
+    specified ``directory``. The program is free to use its own configuration 
+    script or extend the ``.toml`` file. 
+
+    Lastly, ``pip install *.whl`` or ``pip install *.tar.gz`` does the actual
+    installation. If ``*.whl`` is available, ``pip`` will go ahead and copy
+    the files into ``site-packages`` directory. If not, ``pip`` will look at
+    ``pyproject.toml`` and decide what program to use to 'build from source' 
+    (the default is ``setuptools``)
+
+With this standard, switching between packaging tools becomes a lot easier and
+in the case of ``setuptools``, ``setup.py`` becomes optional.
+
+``build_meta`` is ``setuptools``'s implementation of PEP517. It provides the 
+two functions, ``build_wheel`` and ``build_sdist``, amongst others and uses 
+a ``setup.cfg`` to specify the information about the package. 
```

Comment:
The statement "`build_meta` uses a `setup.cfg` to specify information about the package" is not precisely correct. `build_meta` does not stipulate anything about the details of configuring a setuptools package, just provides a PEP-517-compatible hook for building a setuptools package. `setup.cfg` is just one mechanism for configuring a setuptools package.